contrib-compiler.md

```diff
@@ -0,0 +1,99 @@
+
+
+Thanks for interest in contributions.
+
+Note currently our philosphy is to make release build as easy as possible, while dev build is fairly complex at this time.
+
+## Set up dev dependencies
+
+- Having [https://opam.ocaml.org/](opam) installed
+
+    opam switch 4.02.3+buckle-master # use our OCaml compiler
+    opam install camlp4 cppo  <1>
+
+   Both `Camlp4` and `cppo` are used to generate OCaml code for pre-processing. 
+   Switch to our compiler makes your life easier (avoid confusion with diferent compiler versions)
+- Having [https://nodejs.org/](NodeJS) installed
+- Having GNU Make installed
+- OS: Mac/Linux (Note BuckleScript works well on Windows, but the dev mode is not tested)
+- Our dev build depends on a submodule
+
+  ```sh
+  git submodule --update
+  ``` 
+
+All source code are in `jscomp` directory, below assume that you are working in `jscomp` directory.
+
```

Comment:
Wow, this is so great!! Thank you.
Sorry several days went by before my response, I think I missed the notification.